File: js/src/users/Enseignant.js
const Utilisateur = require('./Utilisateur');

/**
 * Classe Enseignant
 * Représente un enseignant qui peut emprunter jusqu'à 5 livres
 */
class Enseignant extends Utilisateur {
    constructor(id, nom, email, departement) {
        super(id, nom, email);
        this.departement = departement;
    }

    static MAX_EMPRUNTS = 5;

    getMaxEmprunt() {
        return Enseignant.MAX_EMPRUNTS;
    }

    getDepartement() {
        return this.departement;
    }

    toString() {
        return `Enseignant{
            id: '${this.id}',
            nom: '${this.nom}',
            email: '${this.email}',
            departement: '${this.departement}',
            livresEmpruntés: ${this.getNombreLivresEmpruntés()}/${Enseignant.MAX_EMPRUNTS}
        }`;
    }
}

module.exports = Enseignant;

File: src/main/java/com/library/users/Enseignant.java
package com.library.users;

/**
 * Classe Enseignant
 * Représente un enseignant qui peut emprunter jusqu'à 5 livres
 */
public class Enseignant extends Utilisateur {
    private String departement;
    private static final int MAX_EMPRUNTS = 5;

    public Enseignant(String id, String nom, String email, String departement) {
        super(id, nom, email);
        this.departement = departement;
    }

    public String getDepartement() {
        return departement;
    }

    @Override
    protected int getMaxEmprunt() {
        return MAX_EMPRUNTS;
    }

    @Override
    public String toString() {
        return "Enseignant{" +
                "id='" + getId() + '\'' +
                ", nom='" + getNom() + '\'' +
                ", email='" + getEmail() + '\'' +
                ", departement='" + departement + '\'' +
                ", livresEmpruntés=" + getNombreLivresEmpruntés() + "/" + MAX_EMPRUNTS +
                '}';
    }
}
